clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -895,14 +895,27 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
     }
 
     if (IntelFPGAII) {
-      unsigned ValueInt = IntelFPGAII->getInterval();
-      if (ValueInt > 0)
-        setSYCLIInterval(ValueInt);
+      unsigned ValueInt = 0;
+      if (auto *E = IntelFPGAII->getIntervalExpr()) {
+        llvm::APSInt ArgVal(32);
+        if (E->isIntegerConstantExpr(ArgVal, Ctx)) {
```

Comment:
This shouldn't be an 'if', it should absolutely be true (since you made sure of it before).  This should assert.  Typically we do something like:

bool Valid = E->IsIntegerConstantExpr(ArgVal, Ctx);
assert(Valid && "Not a ConstantExpr?");(void)Valid;